packages/core/src/sync-historical/service.ts

```diff
@@ -952,115 +939,97 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
-
-    if (!error) return logs;
-
-    const retryRanges: ([Hex, Hex] | readonly [Hex, Hex])[] = [];
-    if (
-      // Alchemy response size error.
-      error.code === InvalidParamsRpcError.code &&
-      error.details!.startsWith("Log response size exceeded.")
-    ) {
-      const safe = error.details!.split("this block range should work: ")[1];
-      const safeStart = Number(safe.split(", ")[0].slice(1));
-      const safeEnd = Number(safe.split(", ")[1].slice(0, -1));
-
-      retryRanges.push([toHex(safeStart), toHex(safeEnd)]);
-      retryRanges.push([toHex(safeEnd + 1), options.toBlock]);
-    } else if (
-      // Another Alchemy response size error.
-      error.details?.includes("Response size is larger than 150MB limit")
-    ) {
-      // No hint available, split into 10 equal ranges.
-      const from = hexToNumber(options.fromBlock);
-      const to = hexToNumber(options.toBlock);
-      const chunks = getChunks({
-        intervals: [[from, to]],
-        maxChunkSize: Math.round((to - from) / 10),
-      });
-      retryRanges.push(
-        ...chunks.map(([f, t]) => [toHex(f), toHex(t)] as const),
-      );
-    } else if (
-      // Infura block range limit error.
-      error.code === LimitExceededRpcError.code &&
-      error.details!.includes("query returned more than 10000 results")
-    ) {
-      const safe = error.details!.split("Try with this block range ")[1];
-      const safeStart = Number(safe.split(", ")[0].slice(1));
-      const safeEnd = Number(safe.split(", ")[1].slice(0, -2));
-
-      retryRanges.push([toHex(safeStart), toHex(safeEnd)]);
-      retryRanges.push([toHex(safeEnd + 1), options.toBlock]);
-    } else if (
-      // Thirdweb block range limit error.
-      error.code === InvalidParamsRpcError.code &&
-      error.details!.includes("block range less than 20000")
-    ) {
-      const midpoint = Math.floor(
-        (Number(options.toBlock) - Number(options.fromBlock)) / 2 +
-          Number(options.fromBlock),
+      const retryRanges = getLogFilterRetryRanges(
+        err as LogFilterError,
+        params.fromBlock,
+        params.toBlock,
       );
 
-      retryRanges.push([toHex(options.fromBlock), toHex(midpoint)]);
-      retryRanges.push([toHex(midpoint + 1), options.toBlock]);
-    } else if (
-      // Quicknode block range limit error (should never happen).
-      error.name === "HttpRequestError" &&
-      error.details!.includes(
-        "eth_getLogs and eth_newFilter are limited to a 10,000 blocks range",
-      )
-    ) {
-      const midpoint = Math.floor(
-        (Number(options.toBlock) - Number(options.fromBlock)) / 2 +
-          Number(options.fromBlock),
-      );
-      retryRanges.push([toHex(options.fromBlock), toHex(midpoint)]);
-      retryRanges.push([toHex(midpoint + 1), options.toBlock]);
-    } else {
-      // Throw any unrecognized errors.
-      throw error;
+      return Promise.all(
+        retryRanges.map(([from, to]) =>
+          this._eth_getLogs({
```

Comment:
Should be fine to `Promise.all` these because concurrency is managed by the queue, right?